From e622bfc49d0f31836c60eb151917f74a0d8ed170 Mon Sep 17 00:00:00 2001
From: Daniel Baluta <daniel.baluta@nxp.com>
Date: Fri, 9 Nov 2018 11:35:46 +0200
Subject: [PATCH 5081/5242] MLK-20189-6: ASoC: fsl: Skip checking for string
 section type

commit  167a6d79f4e34d4d582b3a56ebd0f53988809b8e from
https://source.codeaurora.org/external/imx/linux-imx.git

e_shstrndx already contains the section header index, so
shdr->sh_type will always be SHT_STRTAB.

Remove this redundant check and make Coverity happy.

Fixes: CID3901026
Reviewed-by: Cosmin-Gabriel Samoila <cosmin.samoila@nxp.com>
Signed-off-by: Daniel Baluta <daniel.baluta@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/fsl_dsp.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/sound/soc/fsl/fsl_dsp.c b/sound/soc/fsl/fsl_dsp.c
index d7960b5..b251a18 100644
--- a/sound/soc/fsl/fsl_dsp.c
+++ b/sound/soc/fsl/fsl_dsp.c
@@ -658,8 +658,7 @@ static void dsp_load_firmware(const struct firmware *fw, void *context)
 	shdr = (Elf32_Shdr *)(addr + ehdr->e_shoff +
 			(ehdr->e_shstrndx * sizeof(Elf32_Shdr)));
 
-	if (shdr->sh_type == SHT_STRTAB)
-		strtab = (unsigned char *)(addr + shdr->sh_offset);
+	strtab = (unsigned char *)(addr + shdr->sh_offset);
 
 	/* Load each appropriate section */
 	for (i = 0; i < ehdr->e_shnum; ++i) {
-- 
1.7.9.5

